examples/opensora_hpcai/opensora/models/layers/blocks.py

```diff
@@ -176,15 +176,15 @@ def construct(self, x, cond, mask=None):
         # 2+: mask adaptation for multi-head attention
         if mask is not None:
             # flip mask, since ms FA treats 1 as discard, 0 as retain.
-            mask = 1 - mask
+            mask = 1 - mask.to(ms.int32)
```

Comment:
masks are already `int32`:
https://github.com/mindspore-lab/mindone/blob/7c854c5df7c8324d62af0af8385a50d12076a3ff/examples/opensora_hpcai/opensora/datasets/video_dataset_refactored.py#L191
Instead, fix it in the `t2v_dataset.py`?
https://github.com/mindspore-lab/mindone/blob/7c854c5df7c8324d62af0af8385a50d12076a3ff/examples/opensora_hpcai/opensora/datasets/t2v_dataset.py#L231

Reply:
all mask may be in the same type from dataloader, like int / uint8 / int8 / bool_, remove unnecessary cast and otherwise looks confused